### ISSUE TITLE
iOS: Rebuild is failed

### ISSUE THREAD

**Issue by v0lume** [realm/realm-js#2703]
![image](https://user-images.githubusercontent.com/5075100/73143144-a169c100-40a7-11ea-80e7-674651554bc8.png)

I can rebuild app only after I delete Pods and build folders

## Expected Results

build process is finished and app is running

## Actual Results

building is failed

ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

## Steps to Reproduce

1. react-native run-ios --simulator="iPhone 8"
2. build and run app second time: react-native run-ios --simulator="iPhone 8"

## Version of Realm and Tooling

- Realm JS SDK Version: 3.6.3
- Node or React Native: 0.61.5
- Client OS & Version: macos mojave, iOS
- Which debugger for React Native: None

**Comment by v0lume** [comment]
Also this is not reproduced on previous versions of realm-js

**Comment by bimusiek** [comment]
We have the same issue and only deleting all Pods and re-installing those helps.
However, I did not see this issue re-appearing when setting default node version to 10. 

Can you try with `nvm alias default 10` (if you use `nvm`) or just making sure that your PATH points to node 10 first?

**Comment by v0lume** [comment]
@bimusiek im using v10.16.0 as default

**Comment by olessavluk** [comment]
Got this issue after doing `pod install` with node v12 in `PATH`.

Fixed after doing `rm -rf ios/Pods` and later `pod install`

**Comment by kneth** [comment]
@v0lume I assume that the advices from @bimusiek and @olessavluk fixed your issue. If not, please reopen the issue.

**Comment by EricWiener** [comment]
I keep getting similar errors from RealmJS every time I install a Cocoapod. Whenever I install a Cocoapod, I need to build my project again with XCode. Realm will give me the same errors and I will then need to pod install again. The errors go away the second time. Here's what the errors look like:
<img width="659" alt="Screen Shot 2020-04-07 at 9 06 50 AM" src="https://user-images.githubusercontent.com/18071029/78729846-258ebf00-7909-11ea-824d-9584e507bfc2.png">

**Comment by kyo504** [comment]
I've update realmjs from 3.6.3 to 5.0.3 to support node lts. Unfortunately after this update, I also have been faced with this issue. It works fine if I delete `Pods` folder and reinstall pods but it's pretty annoying especially on ci. Normally I use cached one in order to reduce the build time if there is no change. However I can't use the cache strategy anymore due to this issue.

**Comment by kraenhansen** [comment]
@EricWiener & @kyo504 please open your own issues, this one is closed as it seems the original author got their issue resolved.

**Comment by kyo504** [comment]
@kraenhansen Thanks for your response. I've issued a new ticket and linked this one.

**Comment by zabojad** [comment]
> Got this issue after doing `pod install` with node v12 in `PATH`.
> 
> Fixed after doing `rm -rf ios/Pods` and later `pod install`

This also solved my problem, thanks @olessavluk !
